common/ta_errors.c

```diff
@@ -0,0 +1,158 @@
+#include "common/ta_errors.h"
+
+const char* ta_error_to_string(status_t err) {
+  switch (err) {
+    case SC_OK:
+      return "No error\n";
+    case SC_HTTP_OK:
+      return "HTTP response OK\n";
+    case SC_HTTP_BAD_REQUEST:
+      return "HTTP response, error when parsing request\n";
+    case SC_HTTP_NOT_FOUND:
+      return "HTTP request not found\n";
+    case SC_HTTP_INTERNAL_SERVICE_ERROR:
+      return "HTTP response, other errors in TA\n";
+    case SC_TA_OOM:
+      return "Failed to create TA object\n";
+    case SC_TA_NULL:
+      return "NULL TA objects\n";
+    case SC_TA_WRONG_REQUEST_OBJ:
+      return "Wrong TA request object\n";
+    case SC_TA_LOGGER_INIT_FAIL:
+      return "Failed to init ta logger\n";
+    case SC_TA_ERROR:
+      return "Failed to create cclient object\n";
+    case SC_CCLIENT_OOM:
+      return "Fail to create cclient object\n";
+    case SC_CCLIENT_NOT_FOUND:
+      return "Empty result from cclient\n";
+    case SC_CCLIENT_FAILED_RESPONSE:
+      return "Error in cclient response\n";
+    case SC_CCLIENT_INVALID_FLEX_TRITS:
+      return "flex_trits conversion error\n";
+    case SC_CCLIENT_HASH:
+      return "Hash container operation error\n";
+    case SC_CCLIENT_JSON_KEY:
+      return "JSON key not found\n";
+    case SC_CCLIENT_JSON_PARSE:
+      return "JSON parsing error\n";
+    case SC_CCLIENT_FLEX_TRITS:
+      return "flex_trits converting error\n";
```

Comment:
Please check the difference between `SC_CCLIENT_INVALID_FLEX_TRITS` in line 31. The description is a bit confusing.
Is `SC_CCLIENT_INVALID_FLEX_TRITS` used for checking the validity of `flex_trit` right before the conversion started?

Reply:
I think the error code name is a bit confusing. Take `SC_CCLIENT_FLEX_TRITS` for example, its hard to associate this name with the cause intuitively. Maybe we could rename some error codes more properly in other PR.

By the way, I'll change the description of `SC_CLIENT_INVALID_FLEX_TRITS` to "Invalid flex trits" here.